### ISSUE TITLE
Errors when trying the example with ros foxy/ignition dome

### ISSUE THREAD

**Issue by parkbytes** [AndrejOrsula/ign_moveit2_examples#13]
Hi, thanks for this example. I was trying it out but ran into errors pasted further below.  MoveIt2, Ignition Dome, ros_ign_bridge setup seems to work fine with the standard tutorials/examples in their respective site/repos. 

Environment Details:
Ubuntu: 20.04
ROS Foxy
MoveIt2 - built from source (foxy branch)
Ignition Dome - binary
ros_ign  - built from source to match above combination 

========
Output of ign_moveit2 example_follow_object.launch.py pasted below.  Rviz/Gazebo starts up and displays the model but  rviz warns of missing TF data.  I tried to investigate the Ruckig plugin error and it appears it is not part of MoveIt2 foxy branch.  Not sure if that is an issue.  Wondering if the example is still valid for ros foxy environment.  Appreciate  your response on this. thanks

ubuntu:~/environment/ign_moveit2 $ ros2 launch ign_moveit2 example_follow_object.launch.py
[INFO] [launch]: All log files can be found below /home/ubuntu/.ros/log/2021-12-16-05-02-36-650387-ip-172-31-39-103-74927
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [move_group-1]: process started with pid [74929]
[INFO] [rviz2-2]: process started with pid [74931]
[INFO] [parameter_bridge-3]: process started with pid [74933]
[INFO] [parameter_bridge-4]: process started with pid [74935]
[INFO] [parameter_bridge-5]: process started with pid [74937]
[INFO] [parameter_bridge-6]: process started with pid [74939]
[INFO] [ign gazebo-7]: process started with pid [74941]
[INFO] [parameter_bridge-8]: process started with pid [74944]
[INFO] [example_follow_object.py-9]: process started with pid [74946]
[move_group-1] Parsing robot urdf xml string.
[move_group-1] Link panda_link1 had 1 children
[move_group-1] Link panda_link2 had 1 children
[move_group-1] Link panda_link3 had 1 children
[move_group-1] Link panda_link4 had 1 children
[move_group-1] Link panda_link5 had 1 children
[move_group-1] Link panda_link6 had 1 children
[move_group-1] Link panda_link7 had 1 children
[move_group-1] Link panda_hand had 3 children
[move_group-1] Link panda_link8 had 0 children
[move_group-1] Link panda_leftfinger had 0 children
[move_group-1] Link panda_rightfinger had 0 children
**[move_group-1] [ERROR] [1639630957.622238133] [moveit.ros.occupancy_map_monitor]: No 3D sensor plugin(s) defined for octomap updates**
**[move_group-1] [ERROR] [1639630957.704996049] [moveit.ros_planning.planning_pipeline]: Exception while loading planning adapter plugin 'default_planner_request_adapters/AddRuckigTrajectorySmoothing': According to the loaded plugin descriptions the class default_planner_request_adapters/AddRuckigTrajectorySmoothing with base class type planning_request_adapter::PlanningRequestAdapter does not exist.** Declared types are  chomp/OptimizerAdapter default_planner_request_adapters/AddIterativeSplineParameterization default_planner_request_adapters/AddTimeOptimalParameterization default_planner_request_adapters/AddTimeParameterization default_planner_request_adapters/Empty default_planner_request_adapters/FixStartStateBounds default_planner_request_adapters/FixStartStateCollision default_planner_request_adapters/FixStartStatePathConstraints default_planner_request_adapters/FixWorkspaceBounds default_planner_request_adapters/ResolveConstraintFrames
**[move_group-1] [ERROR] [1639630957.766286646] [moveit.plugins.moveit_simple_controller_manager]: No controller_names specified.**
[move_group-1] Loading 'move_group/ApplyPlanningSceneService'...
[move_group-1] Loading 'move_group/ClearOctomapService'...
[move_group-1] Loading 'move_group/MoveGroupCartesianPathService'...
[move_group-1] Loading 'move_group/MoveGroupExecuteTrajectoryAction'...
[move_group-1] Loading 'move_group/MoveGroupGetPlanningSceneService'...
[move_group-1] Loading 'move_group/MoveGroupKinematicsService'...
[move_group-1] Loading 'move_group/MoveGroupMoveAction'...
[move_group-1] Loading 'move_group/MoveGroupPlanService'...
[move_group-1] Loading 'move_group/MoveGroupQueryPlannersService'...
[move_group-1] Loading 'move_group/MoveGroupStateValidationService'...
[move_group-1] 
[move_group-1] You can start planning now!
[move_group-1] 
[example_follow_object.py-9] [WARN] [1639630958.314876899] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[example_follow_object.py-9] [INFO] [1639630958.457572349] [ign_moveit2_example_follow_object]: ign_moveit2_py initialised successfuly
[rviz2-2] Parsing robot urdf xml string.
[rviz2-2] Link panda_link1 had 1 children
[rviz2-2] Link panda_link2 had 1 children
[rviz2-2] Link panda_link3 had 1 children
[rviz2-2] Link panda_link4 had 1 children
[rviz2-2] Link panda_link5 had 1 children
[rviz2-2] Link panda_link6 had 1 children
[rviz2-2] Link panda_link7 had 1 children
[rviz2-2] Link panda_hand had 3 children
[rviz2-2] Link panda_link8 had 0 children
[rviz2-2] Link panda_leftfinger had 0 children
[rviz2-2] Link panda_rightfinger had 0 children
**[ign gazebo-7] Warning [GenericJoint.hpp:1480] [GenericJoint::setRestPosition] Value of _q0 [0], is out of the limit range [-3.07178, -0.0698132] for index [0] of Joint [Joint].
[move_group-1] [ERROR] [1639630959.227077502] [moveit_robot_state.conversions]: Found empty JointState message
[move_group-1] [ERROR] [1639630959.298389910] [ompl]: /tmp/binarydeb/ros-foxy-ompl-1.5.0/src/ompl/geometric/planners/rrt/src/RRTConnect.cpp:207 - panda_arm/panda_arm: Motion planning start tree could not be initialized!**

**Comment by AndrejOrsula** [comment]
Hello,

Please remove `default_planner_request_adapters/AddRuckigTrajectorySmoothing` from [panda_moveit2_config/launch/move_group_action_server.launch.py#L65](https://github.com/AndrejOrsula/panda_moveit2_config/blob/5d0c9e69a6e2f23e719f3694ff045328084a2aa3/launch/move_group_action_server.launch.py#L65) to get rid of the following error. I expected the error to occur, but I did not expect it to cause any issues and be simply ignored.
```
[move_group-1] [ERROR] [1639630957.704996049] [moveit.ros_planning.planning_pipeline]: Exception while loading planning adapter plugin 'default_planner_request_adapters/AddRuckigTrajectorySmoothing': According to the loaded plugin descriptions the class default_planner_request_adapters/AddRuckigTrajectorySmoothing with base class type planning_request_adapter::PlanningRequestAdapter does not exist. 
```

---

I have not tested this repository with `foxy`/`dome` for quite some time now as I am currently using `rolling`/`fortress`. The following error is especially strange. Please, let me know if the problem (and the error below) persists even after removing the planning adapter.
```
[move_group-1] [ERROR] [1639630957.766286646] [moveit.plugins.moveit_simple_controller_manager]: No controller_names specified.
```

---

On another note, there is https://github.com/AndrejOrsula/ign_moveit2/pull/12 in progress. This PR will deprecate the current approach using `JointTrajectoryController` Ignition plugin and convert the examples into equivalent `ign_ros2_control`+MoveIt 2 examples. It will probably be ready in a few weeks but I don't know when. I will test it on `rolling`/`fortress`, but `foxy`|`galactic`/`edifice` should also work. A bit more info on the current progress is mentioned in https://github.com/AndrejOrsula/ign_moveit2/issues/11#issuecomment-992271711.

**Comment by parkbytes** [comment]
Hi, thanks for your response.  I have tried after removing the planning adapter as suggested but the below error persists.  

```
[move_group-1] [ERROR] [1639711658.446666553] [moveit.ros.occupancy_map_monitor]: No 3D sensor plugin(s) defined for octomap updates
[move_group-1] [ERROR] [1639711658.665691632] [moveit.plugins.moveit_simple_controller_manager]: No controller_names specified.
```

And, no improvement in visualisation outputs.  I continue to see this warning in Rviz. 

```
Fixed Frame
No tf data. Actual error: Frame [panda_link0] does not exist
```

I shall continue my troubleshooting and update if any positive result.  Also, will follow your PR.  Thanks again!

**Comment by sjahr** [comment]
You can ignore ```[moveit.ros.occupancy_map_monitor]: No 3D sensor plugin(s) defined for octomap updates``` This error should not lead to a crash of your setup. 

Regarding the latter error about the non-defined controller names: I am not 100% sure how these parameters are handled within their classes, but there was some recent work on the MoveIt controller manager. You could try to add the required missing parameters despite the fact that ``` 'moveit_manage_controllers': False ``` in [move_group_action_server.launch.py](https://github.com/AndrejOrsula/panda_moveit2_config/blob/5d0c9e69a6e2f23e719f3694ff045328084a2aa3/launch/move_group_action_server.launch.py) like this:

```diff
+   moveit_simple_controllers_yaml = load_yaml(
+        "moveit_resources_panda_moveit_config", "config/panda_controllers.yaml" 
+   )
+    moveit_controllers = {
+        "moveit_simple_controller_manager": moveit_simple_controllers_yaml,
+        "moveit_controller_manager": "moveit_simple_controller_manager/MoveItSimpleControllerManager",
+    }
```
```diff
    # Start the actual move_group node/action server
    run_move_group_node = Node(
        package="moveit_ros_move_group",
        executable="move_group",
        output="screen",
        parameters=[
            robot_description,
            robot_description_semantic,
            kinematics_yaml,
            ompl_planning_pipeline_config,
            trajectory_execution,
+            moveit_controllers,
            planning_scene_monitor_parameters,
            joint_limits_yaml,
        ],
    )
```

**Comment by parkbytes** [comment]
@sjahr , thank you for the response!  

Prior to trying out your suggestion, i upgraded ignition from 'dome' to 'edifice' (binary install) and also  the matching ros_ign package (built from source).  MoveIt2 for foxy was retained as earlier (built from source).  There was no difference in launch errors when trying the example despite this.  I could, however, get the robot to follow the box in ignition gazebo, but for very limited number of moves before the below error appears and robot stops responding to any further box moves in ignition.   

`[move_group-1] [ERROR] [1640049749.667444215] [ompl]: /tmp/binarydeb/ros-foxy-ompl-1.5.0/src/ompl/geometric/planners/rrt/src/RRTConnect.cpp:253 - panda_arm/panda_arm: Unable to sample any valid states for goal tree`

Not sure, but I guess something is messing up during ompl sampling inside MoveIt 2 and reading more about it.

I have since tried out the changes suggested by @sjahr and observed it clears the error related to missing moveit_simple_control_manger.  Also observed that the below error on initialisation no longer appears and the robot model is loaded with box in its hand. (prior to this the gripper was not holding the box at start and below error occured).

`[move_group-1] [ERROR] [1640049631.239364076] [ompl]: /tmp/binarydeb/ros-foxy-ompl-1.5.0/src/ompl/geometric/planners/rrt/src/RRTConnect.cpp:207 - panda_arm/panda_arm: Motion planning start tree could not be initialized! ` 

No improvements in simulation, however - the robot follows the box for few moves and stops responding further with the error below.

`[move_group-1] [ERROR] [1640049749.667444215] [ompl]: /tmp/binarydeb/ros-foxy-ompl-1.5.0/src/ompl/geometric/planners/rrt/src/RRTConnect.cpp:253 - panda_arm/panda_arm: Unable to sample any valid states for goal tree`

There are new errors  saying "action clients not connected" after the introduction of moveit_controllers parameter but I guess those are  expected as the example is not implementing those in its current form.

Happy to say, I am encouraged by your feedback to keep learning! Thanks

**Comment by AndrejOrsula** [comment]
Relevant repositories are now updated to use `ign_ros2_control` instead of `JointTrajectoryController` Ignition plugin. `README.md` is updated with relevant instructions, so I recommend using this approach. I will be closing this issue as it is no longer relevant since the new changes were made. If case you encounter any issues @parkbytes, feel free to open a new issue.

For the sake of completeness, I will give some answers below for the latest errors you were experiencing. Note, that these are no longer relevant in the new version of this repository.

---

Your update from `dome` to `edifice` concludes that the issue here was in fact connected with the utilised version of Ignition. Since `dome` is pretty much EOL now, I won't be looking into it.


The following error appears when a path cannot be planned, e.g. box was moved outside of reach. I am not sure why the node would stop working after that, but it's most likely connected to the utilised MoveIt configuration. If any of your nodes is using `use_sim_time=false`, that could also be causing such issue.
```bash
[move_group-1] [ERROR] [1640049749.667444215] [ompl]: /tmp/binarydeb/ros-foxy-ompl-1.5.0/src/ompl/geometric/planners/rrt/src/RRTConnect.cpp:253 - panda_arm/panda_arm: Unable to sample any valid states for goal tree
```

The `action clients not connected` errors are to be expected if you performed the changes suggested above, since there is no action server present in the old approach. All communication is implemented only via topics. It is possible (and relatively easy) to add action server also for the old approach, but I did not find it necessary when used together with the previous Python interface that already used feedback from the controller.
